### PR TITLE
Chris 1 array data handler

### PR DESCRIPTION
**Pull request by MinceMan**
First step in my master plan. A lot of logic come originally from the ArrayAdapter and then from RecyclerArrayAdapter.

### CODE REVIEW

File: library/src/main/java/com/twotoasters/sectioncursoradapter/adapter/datahandler/ArrayDataHandler.java [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+     * Constructor
+     */
+    public ArrayDataHandler() {
+        init(null, null);
+    }
+
+    /**
+     * Constructor
+     *
+     * @param objects The objects to represent in the ListView. These will be copied so that outside modification is ignored.
```
Comment (patjackson52) [inline library/src/main/java/com/twotoasters/sectioncursoradapter/adapter/datahandler/ArrayDataHandler.java:57]: So modification to the order of the original array will be ignored, but modifications to the objects themselves will still be reflected in the adapter.  is that the desired behavior?

**Comment by patjackson52** [comment]
looking really good.  Glad to see tests!  done with comments

File: library/src/main/java/com/twotoasters/sectioncursoradapter/adapter/datahandler/ArrayDataHandler.java (hunk 1 above)
Comment (MinceMan) [inline library/src/main/java/com/twotoasters/sectioncursoradapter/adapter/datahandler/ArrayDataHandler.java:57]: I think so. I don't really know what I could do about that. I wanted to make sure that changes in the array size are hitting the adapter correctly.

**Comment by MinceMan** [comment]
Addressed comments

**Comment by patjackson52** [comment]
+1

**Pull request merged**